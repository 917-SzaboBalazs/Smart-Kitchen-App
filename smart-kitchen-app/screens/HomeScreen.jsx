import React from 'react';
import { View, Text, Pressable } from 'react-native';
import { StyleSheet } from 'react-native-web';
import colors from '../constants/colors';
import { auth } from '../firebase';
import { signOut } from 'firebase/auth';
import Toast from 'react-native-toast-message';

const HomeScreen = ({ navigation }) => {

    const handleLogOut = () => {
        signOut(auth)
            .then(() => {
                navigation.replace("Home");
            })
            .catch((error) => {
                const errorMessage = error.message;
                //alert(errorMessage);

                Toast.show({
                    type: 'error',
                    text1: 'Error',
                    text2: errorMessage,
                });
            });
    }
    

    return (
        <View style={styles.container}>
            <Text style={styles.mainTitle}>Smart Kitchen App</Text>

            {auth.currentUser &&
                <>
                    <Text style={styles.welcome}>
                        Welcome,
                    </Text>
                    <Text style={styles.userEmail}>
                        {auth.currentUser.email}
                    </Text>
                </>
            }

            <View style={styles.buttonContainer}>
                {auth.currentUser ? 
                    <>
                         <Pressable style={styles.button} onPress={handleLogOut}>
                            <Text style={styles.buttonText}>Log Out</Text>
                        </Pressable>
                    </>
                    :
                    <>
                        <Pressable style={styles.button} onPress={() => navigation.push('Login')}>
                            <Text style={styles.buttonText}>Log In</Text>
                        </Pressable>

                        <Pressable style={styles.button}>
                            <Text style={styles.buttonText} onPress={() => navigation.push('Signup')}>Sign Up</Text>
                        </Pressable>
                    </>
                }

                
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        backgroundColor: colors.background,
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    mainTitle: {
        fontSize: 32,
        fontWeight: 'bold',
        color: colors.primary,
    },
    welcome: {
        fontSize: 20,
        color: colors.primary,
        marginTop: 16,
    },
    userEmail: {
        fontSize: 20,
        color: colors.secondary,
        marginHorizontal: 32,
    },
    buttonContainer: {
        marginTop: 32,
        flexDirection: 'row',
    },
    button: {
        backgroundColor: colors.secondary,
        width: 100,
        height: 40,
        justifyContent: 'center',
        marginRight: 16,
    },
    buttonText: {
        color: 'white',
        textAlign: 'center',
        fontSize: 16,
    },
});

export default HomeScreen;