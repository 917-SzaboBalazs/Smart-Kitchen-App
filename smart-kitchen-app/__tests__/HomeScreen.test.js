import React from "react";
import renderer from "react-test-renderer";
import { ProductContext, ProductProvider } from '../context';
import HomeScreen from "../screens/HomeScreen";

describe("<HomeScreen />", () => {
    it("has 2 element, when not logged in", () => {
        const mockContext = [ [], () => {} ];
        const tree = renderer.create(
            <ProductContext.Provider value={mockContext}>
                <HomeScreen />
            </ProductContext.Provider>
        ).toJSON();
        expect(tree.children.length).toBe(2);
    });

    it("render when logged in", () => {
        const user = "a@a.com";
        const mockContext = [ [], jest.fn(), user, jest.fn()];
        const tree = renderer.create(
            <ProductProvider value={mockContext}>
                <HomeScreen />
            </ProductProvider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});