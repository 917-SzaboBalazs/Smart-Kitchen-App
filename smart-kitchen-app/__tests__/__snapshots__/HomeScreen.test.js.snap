// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> render when logged in 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F7F6BB",
      "flex": 1,
      "justifyContent": "center",
      "paddingVertical": 64,
    }
  }
>
  <Text
    style={
      {
        "color": "#114232",
        "fontSize": 32,
        "fontWeight": "bold",
      }
    }
  >
    Smart Kitchen App
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 32,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#87A922",
          "height": 60,
          "justifyContent": "center",
          "marginRight": 16,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Log In
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#87A922",
          "height": 60,
          "justifyContent": "center",
          "marginRight": 16,
          "width": 100,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "white",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Sign Up
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
